### PR TITLE
build(ci): migrate ci to github actions

### PR DESCRIPTION
**Pull request by brainbicycle**
👋  this wasn't so bad!

Moves build and test ci job from travis to github actions

required some of the fixes from the other branch:
- updates cocoapods to latest to fix installation issues
- updates minimum deployment target in pod and example project for build failures due to missing lib
- deletes crashing test

Let me know if anything needs changing! will update the other PR once this one is merged.

cc @dblock I am not sure why actions are not running here, maybe a repo setting or because I am a first time contributor? 
They ran okay in the fork: https://github.com/artsy/ios-snapshot-test-case-expecta/pull/1

### CODE REVIEW

File: .tool-versions
Comment (brainbicycle) [inline .tool-versions:1]: added this to support asdf / get a consistent ruby version but can remove if not wanted

**Comment by dblock** [comment]
> cc @dblock I am not sure why actions are not running here, maybe a repo setting or because I am a first time contributor?
> They ran okay in the fork: [artsy#1](https://github.com/artsy/ios-snapshot-test-case-expecta/pull/1)

This is GitHub security, it needs a workflow on this repo to exist to be run. It will run after I merge this.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Build and test
+
+on:
+  push:
+    branches:
+      - main
+      - master
```
Comment (dblock) [inline .github/workflows/build-and-test.yml:7]: There's no `main` on this repo, and we want to run it on all branches, consolidate.

```
on: [push, pull_request]
```

File: FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoSpecs.m [hunk 2]
```diff
@@ -282,40 +282,6 @@
             expect(view).toNot.haveValidSnapshotWithTolerance(0.24);
         });
     });
-    
-    it(@"matches view with 256 bytes aligned pointer", ^{
```
Comment (dblock) [inline FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoSpecs.m:286]: I assume this test is a good one other than the crash? Open a new issue for this test crashing so we don't lose it?

File: FBSnapshotTestCaseDemo/Podfile [hunk 3]
```diff
@@ -12,4 +12,12 @@ post_install do |installer|
       config.build_settings['SWIFT_VERSION'] = 4.0
     end
   end
+
+  installer.generated_projects.each do |project|
+    project.targets.each do |target|
+        target.build_configurations.each do |config|
+            config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '14.0'
```
Comment (dblock) [inline FBSnapshotTestCaseDemo/Podfile:19]: Looks like a tab ;) Format file tab/spaces.

**Review by dblock — CHANGES_REQUESTED** [review]
Looks good. Some small asks, plus:

1. Badge in README, `[Build and test](https://github.com/dblock/ios-snapshot-test-case-expecta/actions/workflows/build-and-test.yml)`
2. Add a line to CHANGELOG.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (brainbicycle) [reply .github/workflows/build-and-test.yml:7]: That is what i had initially but with that setting the workflow runs 2x on every push to an open pull request. I can remove `main` for sure.

File: FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoSpecs.m (hunk 2 above)
Comment (brainbicycle) [reply FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoSpecs.m:286]: https://github.com/dblock/ios-snapshot-test-case-expecta/issues/50

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (dblock) [reply .github/workflows/build-and-test.yml]: I would not worry about it, everybody works off forks and these are kinda "separate" runs.

File: .github/workflows/build-and-test.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+name: Build and test
+
+on: [pull_request]
```
Comment (dblock) [inline .github/workflows/build-and-test.yml:3]: Let's put back push, otherwise CI will not run after it's merged to master.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 
 ### 3.x (Next)
 
+* [#49](https://github.com/dblock/ios-snapshot-test-case-expecta/pull/49): Migrate CI to github actions - [@brainbicycle](https://github.com/brainbicycle)
```
Comment (dblock) [inline CHANGELOG.md:4]: Nits: 

1. GitHub is spelled GitHub. You can abbreviate GHA.
2. Put a period at the end of this line.

File: FBSnapshotTestCaseDemo/Podfile [hunk 6]
```diff
@@ -12,4 +12,12 @@ post_install do |installer|
       config.build_settings['SWIFT_VERSION'] = 4.0
     end
   end
+
+  installer.generated_projects.each do |project|
+    project.targets.each do |target|
+        target.build_configurations.each do |config|
```
Comment (dblock) [inline FBSnapshotTestCaseDemo/Podfile:18]: Looks like a tab in front of target, should be two spaces.

**Review by dblock — CHANGES_REQUESTED** [review]
Almost there. Thanks for hanging in here with me.

**Pull request merged**